From db54d2f903856adbd5763dbeaaf6407a9328c8e2 Mon Sep 17 00:00:00 2001
From: DennySPb <dennyspb@gmail.com>
Date: Wed, 14 Oct 2020 09:29:06 +0300
Subject: [PATCH] FaceUnlock: Drop lockouts completely

 * Our face unlock service is a little buggy. After a
   night standby we see an excess of attempts, although
   in fact there were none at all. Let's just remove
   lockouts completely.

Change-Id: Ie41d7fd12274c6abf714a5fbab862146f65a6de3
---
 .../android/server/biometrics/face/FaceService.java   | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/services/core/java/com/android/server/biometrics/face/FaceService.java b/services/core/java/com/android/server/biometrics/face/FaceService.java
index 0d99a707783d..3ad1bb69ba50 100644
--- a/services/core/java/com/android/server/biometrics/face/FaceService.java
+++ b/services/core/java/com/android/server/biometrics/face/FaceService.java
@@ -972,14 +972,7 @@ public void onEnumerate(long deviceId, ArrayList<Integer> faceIds, int userId)
         @Override
         public void onLockoutChanged(long duration) {
             Slog.d(TAG, "onLockoutChanged: " + duration);
-            if (duration == 0) {
-                mCurrentUserLockoutMode = AuthenticationClient.LOCKOUT_NONE;
-            } else if (duration == Long.MAX_VALUE) {
-                mCurrentUserLockoutMode = AuthenticationClient.LOCKOUT_PERMANENT;
-            } else {
-                mCurrentUserLockoutMode = AuthenticationClient.LOCKOUT_TIMED;
-            }
-
+            mCurrentUserLockoutMode = AuthenticationClient.LOCKOUT_NONE;
             mHandler.post(() -> {
                 if (duration == 0) {
                     notifyLockoutResetMonitors();
@@ -1305,7 +1298,7 @@ protected int statsModality() {
 
     @Override
     protected int getLockoutMode() {
-        return mCurrentUserLockoutMode;
+        return AuthenticationClient.LOCKOUT_NONE;
     }
 
     /** Gets the face daemon */
